Config/CMakeIncludes/sanity_check_and_xsdk.cmake

```diff
@@ -174,20 +177,42 @@ if (Tasmanian_ENABLE_MPI)
     endif()
 endif()
 
-# check for fortran, note that enable_language always gives FATAL_ERROR if compiler is missing (no way to recover or autodisable)
+# check for Fortran, note that enable_language always gives FATAL_ERROR if the compiler is missing (no way to recover or autodisable)
 if (Tasmanian_ENABLE_FORTRAN)
     enable_language(Fortran)
 endif()
 
-# debug flags and workarounds (Fortran doesn't know how to handle those)
-if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-    add_definitions(-D_TASMANIAN_DEBUG_)
-    if (Tasmanian_ENABLE_FORTRAN) # AND (${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU"))
-        set(CMAKE_Fortran_FLAGS_DEBUG "-g")
-    endif()
-else()
-    # this is a bug in some versions of CUDA
-    if ((Tasmanian_ENABLE_CUBLAS OR Tasmanian_ENABLE_CUDA) AND ("${CUDA_VERSION_MAJOR}" LESS 8))
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_FORCE_INLINES")
+
+########################################################################
+# Compiler specific flags: Intel hasn't been tested in a while
+# Tasmanian_STRICT_OPTIONS=ON will prevent Tasmanian from setting
+# compiler flags, only the user provided flags will be used
+########################################################################
+if (NOT Tasmanian_STRICT_OPTIONS)
```

Comment:
I don't know how I feel about these, especially with default `Tasmanian_STRICT_OPTIONS` is `OFF`.

Reply:
This is done for low level users that don't want to read messages. We can default to `ON` and set it in the install script.

There are almost no flags, just bug work-around. The strict options affects what happens if `find_package(BLAS)` or `find_package(CUDA)` fail.

XSDK sets this to on.